content/kotlin/concepts/data-types/data-types.md

```diff
@@ -0,0 +1,365 @@
+---
+Title: 'Data Types'
+Description: 'Data types are a classification of types of values that can be stored and manipulated in a program.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Primitive Data-Type'
+  - 'Non Primitive Data-Type'
+  
+CatalogContent:
+  - 'learn-Kotlin'
+  - 'paths/computer-science'
+---
+
+**Data types** are a classification of types of data that determine the possible values and operations that can be performed on that data.
+
+## Type of Data-Types 
+
+Kotlin has both primitive and non-primitive data types.
+1. Primitive Data types
+2. Non-primitive Data types 
+
+## Primitive Data-type
+
+There are the following different types of primitive data types in Kotlin:
+
+- Boolean
+- Byte
+- Char
+- Double
+- Float
+- Int
+- Long
+- Short
+
+### Boolean 
+
+In Kotlin, the `Boolean` is a primitive data type and can hold only one of the two possible values: `true` or `false`.
+
+Here is an example of how to declare a `Boolean` variable in Kotlin:
+
+```kotlin
+val isTrue: Boolean = true
+val isFalse: Boolean = false
+// here we can see that if a variable is wrong we can written false and true if its right
+```
+
+The `Boolean` data type can be used to determine the outcome of `if ... then` statements:
+
+```kotlin
+fun main(){
+    val condition: Boolean = true
+
+    if (condition) {
+        // write a code that you want to execute if the variable is given as true 
+    } else {
+        // write a code for the false variable 
+    }
+}
+```
+
+### Byte
+
+A `Byte` is a data type that is similar to an integer data type, but it is only 8 bits in size, whereas an integer can be a different size depending on the system. A byte can hold values from -128 to 127.
+
+A `Byte` is a way to represent a small whole number. It is commonly used to store small pieces of data, such as characters in a text file or colors in an image.
+
+Here is an example of how to declare a `Byte` variable in Kotlin:
+
+```kotlin
+fun main(){
+    val a: Byte = 50
+    val b: Byte = 25
+    val c: Byte = a + b as Byte
+    print(c)
+
+}
+```
+
+The output will be:
+
+```
+75
+```
+
+The range of the `Byte` data type is -128 to 127. Calculations that exceed this range will produce unexpected results:
+
+```kotlin
+fun main(){
+    val a: Byte = 120
+    val b: Byte = 120
+    val c: Byte = a + b as Byte
+    print(c)
+}
+```
+Here the output will be:
```

Comment:
There need to be line breaks before and after the code blocks.

Reply:
got it

Reply:
fixed it @SSwiniarski